client/web/src/nav/NavLinks.tsx

```diff
@@ -158,9 +158,11 @@ export const NavLinks: React.FunctionComponent<Props> = props => {
                     </li>
                 ))}
             {/* show status messages if authenticated user is admin or opted-in with a user tag  */}
-            {(authenticatedUser?.siteAdmin || authenticatedUser?.tags?.includes('AllowUserExternalServicePublic')) && (
+            {(window.context.externalServicesUserModeEnabled ||
+                authenticatedUser?.siteAdmin ||
+                authenticatedUser?.tags?.includes('AllowUserExternalServicePublic')) && (
```

Comment:
I think just always checking an authenticated user exists first should help :)
```suggestion
            {(authenticatedUser && (window.context.externalServicesUserModeEnabled ||
                authenticatedUser.siteAdmin ||
                authenticatedUser.tags?.includes('AllowUserExternalServicePublic'))) && (
```

Reply:
Ah, didn't see your messages. I just pushed something similar to @eseliger's solution